includes/cli/class-amp-cli-validation-command.php

```diff
@@ -314,6 +314,69 @@ public function reset( $args, $assoc_args ) {
 		WP_CLI::success( 'All AMP validation data has been removed.' );
 	}
 
+	/**
+	 * Generate the authorization nonce needed for a validate request.
+	 *
+	 * @subcommand generate-nonce
+	 * @alias nonce
+	 */
+	public function generate_nonce() {
+		WP_CLI::line( AMP_Validation_Manager::get_amp_validate_nonce() );
+	}
+
+	/**
+	 * Get the validation results for a given URL.
+	 *
+	 * The results are returned in JSON format.
+	 *
+	 * ## OPTIONS
+	 *
+	 * <url>
+	 * : The URL to check. The host name need not be included. The URL must be local to this WordPress install.
+	 *
+	 * ## EXAMPLES
+	 *
+	 *     wp amp validation check-url /about/
+	 *     wp amp validation check-url $( wp option get home )/?p=1
+	 *
+	 * @subcommand check-url
+	 * @alias check
+	 *
+	 * @param array $args Args.
+	 */
+	public function check_url( $args ) {
+		list( $url ) = $args;
+
+		$host            = wp_parse_url( $url, PHP_URL_HOST );
+		$parsed_home_url = wp_parse_url( home_url( '/' ) );
+
+		if ( $host && $host !== $parsed_home_url['host'] ) {
+			WP_CLI::error(
+				sprintf(
+					/* translators: %1$s is the expected host, %2$s is the actual host */
+					__( 'Supplied URL must be for this WordPress install. Expected host "%1$s" but provided is "%2$s".', 'amp' ),
```

Comment:
WP-CLI output is generally not translated, to keep it easily parseable by scripts.